### PR TITLE
Warn about using old minor versions of Scala.

### PR DESCRIPTION
**Pull request by leifwickland**
Try to help others avoid the pain we're seeing in #292.

### CODE REVIEW

**Comment by ruippeixotog** [comment]
If #292 is found out to be a real problem with some patch versions this LGTM, but as I mentioned [here](https://github.com/pureconfig/pureconfig/issues/292#issuecomment-320111390) I think we should investigate the issue a bit further.

**Comment by leifwickland** [comment]
@ruippeixotog Yup. I'm totally OK with that. I'm already beyond my understanding of PureConfig in #292, so I'll leave it to more capable hands.

File: core/src/main/tut/README.md [hunk 1]
```diff
@@ -88,6 +88,7 @@ libraryDependencies ++= Seq(
 For a full example of `build.sbt` you can have a look at this [build.sbt](https://github.com/melrief/pureconfig/blob/master/example/build.sbt)
 used for the example.
 
+Earlier versions of Scala had bugs which can cause subtle compile-time problems in PureConfig. As a result we recommend only using 2.11.9 or 2.12.1 or newer within the minor series.
```
Comment (ruippeixotog) [inline core/src/main/tut/README.md]: 2.11.9 is a "forbidden" version :) Both 2.11.9 and 2.11.10 were intended to be the last version in the 2.11.x series, but they both introduced binary compatibility issues and were quickly superseded by 2.11.11. for that reason, I think we should recommend using 2.11.11.

See https://github.com/scala/scala/releases/tag/v2.11.11 for more details.

**Comment by melrief** [comment]
I think having this in the README can help developers. I agree with the content of the message regardless of #292, last versions of Scala should be much better for libraries using shapeless, like our.

**Review by ruippeixotog — APPROVED** [review]
LGTM!

File: core/src/main/tut/README.md (hunk 1 above)
Comment (leifwickland) [reply core/src/main/tut/README.md]: Good point, @ruippeixotog. Updated.

**Pull request merged**